### PR TITLE
Preliminary steps for using scala 3

### PR DESCRIPTION
**Pull request by jtjeferreira**
The upgrade to scala 3 requires upgrading a few libraries to versions that have support for it. In this PR:
* scalatest is upgraded from 3.0.x to 3.2.x
* scalamock is replaced by the use of mockito directly since scalamock does yet support scala3 (see https://github.com/paulbutcher/ScalaMock/issues/429)
* scalafmt is upgraded from 2.x to 3.x trying to keep the old scalafmt configuration, and reducing the number of formatting changes.

### CODE REVIEW

File: bson-scala/src/main/scala-2.13+/org/mongodb/scala/bson/collection/mutable/Document.scala [hunk 1]
```diff
@@ -140,7 +140,7 @@ case class Document(protected[scala] val underlying: BsonDocument)
   @inline final def ++(suffix: IterableOnce[(String, BsonValue)]): Document = concat(suffix)
   // scalastyle:on method.name
   def map[B](f: ((String, BsonValue)) => (String, BsonValue)): Document = strictOptimizedMap(newSpecificBuilder, f)
-  //TODO other operations
+  // TODO other operations
```
Comment (jtjeferreira) [inline bson-scala/src/main/scala-2.13+/org/mongodb/scala/bson/collection/mutable/Document.scala:143]: scalafmt now adds a space to comments. I could not find an option to disable this behavior.

File: bson-scala/src/main/scala/org/mongodb/scala/bson/codecs/macrocodecs/CaseClassCodec.scala [hunk 2]
```diff
@@ -127,18 +127,15 @@ private[codecs] object CaseClassCodec {
 
     val fields: Map[Type, List[(TermName, Type)]] = {
       knownTypes
-        .map(
```
Comment (jtjeferreira) [inline bson-scala/src/main/scala/org/mongodb/scala/bson/codecs/macrocodecs/CaseClassCodec.scala:130]: I think this change is okay, so I didnt look to much in finding an option to revert this change

File: bson-scala/src/test/scala/org/mongodb/scala/bson/BaseSpec.scala [hunk 3]
```diff
@@ -16,8 +16,9 @@
...
 
 import org.junit.runner.RunWith
-import org.scalatest.{ FlatSpec, Matchers }
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
 import org.scalatestplus.junit.JUnitRunner
 
 @RunWith(classOf[JUnitRunner])
-abstract class BaseSpec extends FlatSpec with Matchers {}
+abstract class BaseSpec extends AnyFlatSpec with Matchers {}
```
Comment (jtjeferreira) [inline bson-scala/src/test/scala/org/mongodb/scala/bson/BaseSpec.scala:24]: AnyFlatSpec is the new name of `FlatSpec`. See https://www.scalatest.org/release_notes/3.1.0

File: build.gradle [hunk 4]
```diff
@@ -30,7 +30,7 @@ buildscript {
 
         // Scala plugins
         classpath "com.adtran:scala-multiversion-plugin:2.0.4"
-        classpath "com.diffplug.spotless:spotless-plugin-gradle:3.27.1"
+        classpath "com.diffplug.spotless:spotless-plugin-gradle:6.3.0"
```
Comment (jtjeferreira) [inline build.gradle:33]: upgrading spotless plugin was necessary to use a more recent scalafmt

File: build.gradle [hunk 5]
```diff
@@ -120,7 +122,7 @@ configure(scalaProjects) {
 
     spotless {
         scala {
-            scalafmt('2.0.0').configFile("$configDir/scala/scalafmt.conf")
```
Comment (jtjeferreira) [inline build.gradle:123]: the version of scalafmt to use is now configured in in `scalafmt.conf`

File: driver-scala/src/test/scala/org/mongodb/scala/AggregateObservableSpec.scala [hunk 6]
```diff
@@ -49,17 +49,6 @@ class AggregateObservableSpec extends BaseSpec with MockFactory {
     val ct = classOf[Document]
     val verbosity = ExplainVerbosity.QUERY_PLANNER
 
-    wrapper.expects(Symbol("allowDiskUse"))(true).once()
```
Comment (jtjeferreira) [inline driver-scala/src/test/scala/org/mongodb/scala/AggregateObservableSpec.scala:52]: the style of the test changed a bit, but was done in a semi automated way. I managed to use a regex to transform
```scala
 wrapper.expects(Symbol("allowDiskUse"))(true).once()
```
into
```scala
verify(wrapper).allowDiskUse(true)
```
and then move the lines to "after" the test instead of before

**Comment by jtjeferreira** [comment]
Hi @rozza. 

Do you think this could get a review?

**Comment by rozza** [comment]
Hi @jtjeferreira,

Many thanks for the PR - I used the commits and rebased into master.

See: 

e9584dd1a75ccbce9974fc3751ab24d4cc537d81
7a0b575dbb7788abfc7e537e31fe7c7b49b2dac5
8977afabdada7ac67340e291017cbc1355db5b9b
